packages/stock-location/src/utils/build-query.ts

```diff
@@ -159,7 +159,11 @@ function buildWhere<TWhereKeys extends object, TEntity>(
       continue
     }
 
-    where[key] = value
+    if (key === "q") {
+      where["name"] = ILike(`%${value}%`)
+    } else {
+      where[key] = value
+    }
```

Comment:
Would it perhaps make sense to instead do like we do in `medusa`:

```
let q: string | undefined

if (selector.q) {
  q = selector.q
  delete selector.q
}

const query = buildQuery(selector, config)

if (q) {
  const where = query.where as ...

  delete where.name

  query.where = [
    {
      ...where,
      name: ILike(`%${q}%`)
    }
  ]
}

// ... find and count
```

Otherwise if you have `?q=something&name=something_else` then `q` will be overridden, and we usually have the opposite where `q` overrides any filter set on the searchable fields.